GooglePlayInstant/Editor/AndroidManifestHelper.cs

```diff
@@ -0,0 +1,241 @@
+            XElement defaultUrlMetaData;
+            switch (metaDataElements.Count())
+            {
+                case 0:
+                    defaultUrlMetaData = new XElement(MetaData);
+                    mainActivity.Add(defaultUrlMetaData);
+                    break;
+                case 1:
+                    defaultUrlMetaData = metaDataElements.First();
+                    defaultUrlMetaData.RemoveAttributes();
+                    break;
+                default:
+                    return "more than one meta-data element for default-url";
+            }
+
+            defaultUrlMetaData.SetAttributeValue(AndroidNameXName, DefaultUrl);
+            defaultUrlMetaData.SetAttributeValue(AndroidValueXName, uri);
+
+            return null;
+        }
+
+        private static IEnumerable<XElement> GetMainActivities(XContainer applicationElement)
+        {
+            // See https://developer.android.com/topic/instant-apps/getting-started/prepare.html#default-url
+            // Find all activities with an <intent-filter> that has
+            //  <action android:name="android.intent.action.MAIN" />
+            //  <category android:name="android.intent.category.LAUNCHER" />
+            return
+                from activityElement in applicationElement.Elements(Activity)
+                where
+                    (from intentFilter in activityElement.Elements(IntentFilter)
+                        where
+                            intentFilter.Elements(Action)
+                                .Any(e => (string) e.Attribute(AndroidNameXName) == ActionMain) &&
+                            intentFilter.Elements(Category)
+                                .Any(e => (string) e.Attribute(AndroidNameXName) == CategoryLauncher)
+                        select intentFilter)
+                    .Any()
+                select activityElement;
+        }
+
+        private static IEnumerable<XElement> GetActionViewIntentFilters(XContainer mainActivity)
+        {
+            return from intentFilter in mainActivity.Elements(IntentFilter)
+                where intentFilter.Elements(Action).Any(e => (string) e.Attribute(AndroidNameXName) == ActionView)
+                select intentFilter;
+        }
+
+        private static IEnumerable<XElement> GetDefaultUrlMetaDataElements(XContainer mainActivity)
+        {
+            return from metaData in mainActivity.Elements(MetaData)
+                where (string) metaData.Attribute(AndroidNameXName) == DefaultUrl
+                select metaData;
+        }
+
+        private static XElement CreateElementWithAttribute(string elementName, XName attributeName,
+            string attributeValue)
+        {
+            var element = new XElement(elementName);
+            element.SetAttributeValue(attributeName, attributeValue);
+            return element;
+        }
+
+        private static XElement GetExactlyOne(IEnumerable<XElement> elements)
```

Comment:
FirstOrDefault?
https://stackoverflow.com/questions/497261/how-do-i-get-the-first-element-from-an-ienumerablet-in-net
https://msdn.microsoft.com/en-us/library/bb340482(v=vs.110).aspx